### ISSUE TITLE
Frozen surface/mesh dictionary setting

### ISSUE THREAD

**Issue by jexalto** [mdolab/OpenAeroStruct#415]
# Description of feature
Check whether keys already exist in the default dicitonary. return error when a key in the dictionary doesn't exist in the default dictionary. this way we prevent the addition fo keys+values that don't exist. For example, this will raise an error when you add variables that don’t exist in the default dict. This adds a little more robustness than being able to accidentally add variables in current OAS.

# Potential solution
```python
class AttrDict(dict):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.__dict__ = self

    def __setattr__(self, key, value):
        if key not in [*self.keys(), '__dict__']:
            raise KeyError('No new keys allowed')
        else:
            super().__setattr__(key, value)

    def __setitem__(self, key, value):
        if key not in self:
            raise KeyError('No new keys allowed')
        else:
            super().__setitem__(key, value)
```

**Comment by kanekosh** [comment]
I think this would be a nice feature, but I'd hesitate to raise an error because that would potentially be backward incompatible. I'd suggest raising a warning instead of an error.

**Comment by kanekosh** [comment]
Fixed by #417
